rust/lance-encoding/src/decoder.rs

```diff
@@ -813,18 +825,30 @@ impl CoreFieldDecoderStrategy {
                             false,
                         )))
                     };
-                    let list_field = Field::try_from(ArrowField::new(
+                    let list_field_result = Field::try_from(ArrowField::new(
                         field.name.clone(),
                         list_type,
                         field.nullable,
-                    ))
-                    .unwrap();
-                    let list_scheduler = self.create_list_scheduler(
+                    ));
+
+                    if list_field_result.is_err() {
+                        return Err(list_field_result.err().unwrap());
+                    }
+
+                    let list_field = list_field_result.unwrap();
```

Comment:
Why not just use `?` here?
```suggestion
                    let list_field = Field::try_from(ArrowField::new(
                        field.name.clone(),
                        list_type,
                        field.nullable,
                    ))?;
```